src/components/Context.js

```diff
@@ -183,49 +219,51 @@ class ProductProvider extends Component {
   addTotals = () => {
     let subTotal = 0;
     this.state.cart.map((item) => (subTotal += item.total));
-    const tempTax = subTotal * 0.1;
-    const tax = parseFloat(tempTax.toFixed(2));
-    const total = subTotal + tax;
+    const total = subTotal;
+
     this.setState(() => {
       return {
         cartSubTotal: subTotal,
-        cartTax: tax,
         cartTotal: total,
       };
     });
+    this.addPromotion();
   };
 
-  addPromotionToCart = (couponCode) => {
-    client.vouchers.get(couponCode, (error, result) => {
-      if (error) {
-        toast.error('Promotion not found');
-        return error;
-      }
-      const voucher = result;
-      console.log(voucher);
+  addPromotionToCart = async (couponCode) => {
+    try {
+      const voucher = await new Promise((resolve, reject) => {
+        window.Voucherify.validate(couponCode, (response) => {
+          if (response.valid) {
+            resolve(response);
+          } else {
+            reject(new Error(response.reason));
+          }
+        });
+      });
+
       this.setState(
         () => {
           return {
-            promotionItems: voucher,
-            promotionItemsNumber: 1,
-            appliedCouponCode: couponCode,
+            appliedVoucher: voucher,
           };
         },
         () => {
           toast.success('Promotion applied');
-          this.countDiscount();
+          this.checkDiscount();
+          this.addPromotion();
```

Comment:
once you move the logic of these two functions into calculateCart() reducer function, these two calls will be deleted.